### PR TITLE
Disallow boolean `negative`

### LINKED ISSUE #692: Negative operation for a boolean array is forbidden in NumPy
In NumPy negative operation for a boolean array is forbidden.
```
>>> -numpy.array([True])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: The numpy boolean negative, the `-` operator, is not supported, use the `~` operator or the logical_not function instead.
>>> -cupy.array([True])
array([False], dtype=bool)
```

### PR DESCRIPTION
**Pull request by toslunar**
Close #692.

### CODE REVIEW

File: cupy/core/_kernel.pyx [hunk 1]
```diff
@@ -1115,16 +1115,18 @@ cdef class _Ops:
...
-            if isinstance(t, _Op):
-                ops_.append(t)
-            elif isinstance(t, tuple):
+            if isinstance(t, tuple):
                 typ, rt = t
                 if isinstance(rt, tuple):
                     rt = tuple([r1 or r2 for r1, r2 in zip(rt, routine)])
-                ops_.append(_Op.from_type_and_routine(typ, rt))
+                elif not isinstance(rt, str):
+                    # TODO(kataoka): assert isinstance(rt, callable)
```
Comment (niboshi) [inline cupy/core/_kernel.pyx]: Why is this TODO? (It should be `assert callable(rt)` actually)

Comment (toslunar) [reply cupy/core/_kernel.pyx]: It seems `assert callable(rt)` cannot be done in `cdef`.

Comment (niboshi) [reply cupy/core/_kernel.pyx]: Do you mean `assert` or `callable`, or their combination?
`assert` is used in the code already and I personally confirmed `callable` is usable in Cython.

Comment (niboshi) [reply cupy/core/_kernel.pyx]: `assert callable(obj)` seems to work, too.

Comment (toslunar) [reply cupy/core/_kernel.pyx]: Sorry.  `assert callable(obj)` works.

**Comment by niboshi** [comment]
Jenkins test this please

**Pull request merged**